pkg/api/osa/vlabs/types.go

```diff
@@ -70,6 +78,14 @@ const (
 	Upgrading ProvisioningState = "Upgrading"
 )
 
+type CredentialProfile struct {
+	// I believe the same token can be used for secured registry and for
+	// Insights.
+	RedHatCustomerPortalUsername string `json:"redHatCustomerPortalUsername,omitempty"`
+	RedHatCustomerPortalPassword string `json:"redHatCustomerPortalPassword,omitempty"`
+	// we can add a KeyVaultRef here later.
+}
```

Comment:
Should this be a slice? Will registry, insights, in the future node auto updated callbacks will use same passwords?